LWC Module threw an uncaught exception! LWC version: 5.0.16

[22:56:31] [Server thread/WARN]: com.griefcraft.scripting.ModuleException: LWC Module threw an uncaught exception! LWC version: 5.0.16
[22:56:31] [Server thread/WARN]:        at com.griefcraft.scripting.ModuleLoader.dispatchEvent(ModuleLoader.java:351)
[22:56:31] [Server thread/WARN]:        at com.griefcraft.listeners.LWCEntityListener.entityCreatedByPlayer(LWCEntityListener.java:182)
[22:56:31] [Server thread/WARN]:        at com.griefcraft.listeners.LWCEntityListener.onHangingPlace(LWCEntityListener.java:91)
[22:56:31] [Server thread/WARN]:        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[22:56:31] [Server thread/WARN]:        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[22:56:31] [Server thread/WARN]:        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[22:56:31] [Server thread/WARN]:        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[22:56:31] [Server thread/WARN]:        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306)
[22:56:31] [Server thread/WARN]:        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[22:56:31] [Server thread/WARN]:        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589)
[22:56:31] [Server thread/WARN]:        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576)
[22:56:31] [Server thread/WARN]:        at net.minecraft.world.item.ItemHanging.a(ItemHanging.java:72)
[22:56:31] [Server thread/WARN]:        at net.minecraft.world.item.ItemStack.useOn(ItemStack.java:279)
[22:56:31] [Server thread/WARN]:        at net.minecraft.server.level.PlayerInteractManager.a(PlayerInteractManager.java:552)
[22:56:31] [Server thread/WARN]:        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1547)
[22:56:31] [Server thread/WARN]:        at net.minecraft.network.protocol.game.PacketPlayInUseItem.a(PacketPlayInUseItem.java:33)
[22:56:31] [Server thread/WARN]:        at net.minecraft.network.protocol.game.PacketPlayInUseItem.a(PacketPlayInUseItem.java:1)
[22:56:31] [Server thread/WARN]:        at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:31)
[22:56:31] [Server thread/WARN]:        at net.minecraft.server.TickTask.run(SourceFile:18)
[22:56:31] [Server thread/WARN]:        at net.minecraft.util.thread.IAsyncTaskHandler.d(SourceFile:157)
[22:56:31] [Server thread/WARN]:        at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(SourceFile:23)
[22:56:31] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1190)
[22:56:31] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:1)
[22:56:31] [Server thread/WARN]:        at net.minecraft.util.thread.IAsyncTaskHandler.y(SourceFile:131)
[22:56:31] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.be(MinecraftServer.java:1169)
[22:56:31] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.y(MinecraftServer.java:1162)
[22:56:31] [Server thread/WARN]:        at net.minecraft.util.thread.IAsyncTaskHandler.bo(SourceFile:116)
[22:56:31] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1145)
[22:56:31] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1051)
[22:56:31] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:304)
[22:56:31] [Server thread/WARN]:        at java.base/java.lang.Thread.run(Thread.java:833)
[22:56:31] [Server thread/WARN]: Caused by: java.lang.NullPointerException: Cannot invoke "org.bukkit.block.Block.getLocation()" because the return value of "com.griefcraft.model.Protection.getBlock()" is null
[22:56:31] [Server thread/WARN]:        at de.robotricker.transportpipes.utils.LWCUtils.onPostRegistration(LWCUtils.java:40)
[22:56:31] [Server thread/WARN]:        at com.griefcraft.scripting.ModuleLoader.dispatchEvent(ModuleLoader.java:343)
[22:56:31] [Server thread/WARN]:        ... 30 more

[22:17:03] [Server thread/ERROR]: Error occurred while enabling Bookshelf v3.0.8.8 (Is it up to date?)
java.lang.NoClassDefFoundError: com/griefcraft/scripting/event/LWCEntityInteractEvent
        at java.lang.Class.getDeclaredMethods0(Native Method) ~[?:?]
        at java.lang.Class.privateGetDeclaredMethods(Class.java:3402) ~[?:?]
        at java.lang.Class.getDeclaredMethods(Class.java:2504) ~[?:?]
        at com.griefcraft.scripting.ModuleLoader.doesObjectOverrideMethod(ModuleLoader.java:299) ~[?:?]
        at com.griefcraft.scripting.ModuleLoader.registerFastCache(ModuleLoader.java:218) ~[?:?]
        at com.griefcraft.scripting.ModuleLoader.registerModule(ModuleLoader.java:460) ~[?:?]
        at com.loohp.bookshelf.listeners.hooks.LWCEvents.hookLWC(LWCEvents.java:54) ~[?:?]
        at com.loohp.bookshelf.Bookshelf.onEnable(Bookshelf.java:298) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:342) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:480) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_18_R2.CraftServer.enablePlugin(CraftServer.java:518) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3493-Spigot-b081915-9d46b56]
        at org.bukkit.craftbukkit.v1_18_R2.CraftServer.enablePlugins(CraftServer.java:432) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3493-Spigot-b081915-9d46b56]
        at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:612) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3493-Spigot-b081915-9d46b56]
        at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:414) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3493-Spigot-b081915-9d46b56]
        at net.minecraft.server.dedicated.DedicatedServer.e(DedicatedServer.java:263) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3493-Spigot-b081915-9d46b56]
        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1007) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3493-Spigot-b081915-9d46b56]
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:304) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3493-Spigot-b081915-9d46b56]
        at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.ClassNotFoundException: com.griefcraft.scripting.event.LWCEntityInteractEvent
        at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:147) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:99) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
        ... 19 more

--- a/src/main/java/com/loohp/bookshelf/listeners/hooks/LWCEvents.java
+++ b/src/main/java/com/loohp/bookshelf/listeners/hooks/LWCEvents.java
@@ -24,23 +24,11 @@ import com.griefcraft.lwc.LWC;
 import com.griefcraft.model.Permission.Access;
 import com.griefcraft.model.Protection;
 import com.griefcraft.model.Protection.Type;
-import com.griefcraft.scripting.Module;
+import com.griefcraft.scripting.JavaModule;
 import com.griefcraft.scripting.event.LWCAccessEvent;
-import com.griefcraft.scripting.event.LWCBlockInteractEvent;
-import com.griefcraft.scripting.event.LWCCommandEvent;
-import com.griefcraft.scripting.event.LWCDropItemEvent;
-import com.griefcraft.scripting.event.LWCEntityInteractEvent;
-import com.griefcraft.scripting.event.LWCMagnetPullEvent;
-import com.griefcraft.scripting.event.LWCProtectionDestroyEvent;
-import com.griefcraft.scripting.event.LWCProtectionInteractEntityEvent;
 import com.griefcraft.scripting.event.LWCProtectionInteractEvent;
-import com.griefcraft.scripting.event.LWCProtectionRegisterEntityEvent;
 import com.griefcraft.scripting.event.LWCProtectionRegisterEvent;
-import com.griefcraft.scripting.event.LWCProtectionRegistrationPostEvent;
 import com.griefcraft.scripting.event.LWCProtectionRemovePostEvent;
-import com.griefcraft.scripting.event.LWCRedstoneEvent;
-import com.griefcraft.scripting.event.LWCReloadEvent;
-import com.griefcraft.scripting.event.LWCSendLocaleEvent;
 import com.loohp.bookshelf.Bookshelf;
 import com.loohp.bookshelf.api.events.PlayerOpenBookshelfEvent;
 import com.loohp.bookshelf.objectholders.BookshelfHolder;
@@ -48,20 +36,10 @@ import com.loohp.bookshelf.objectholders.LWCRequestOpenData;
 import org.bukkit.Bukkit;
 import org.bukkit.entity.Player;
 
-public class LWCEvents implements Module {
+public class LWCEvents extends JavaModule {
 
     public static void hookLWC() {
         LWC.getInstance().getModuleLoader().registerModule(Bookshelf.plugin, new LWCEvents());
-    }
-
-    @Override
-    public void onReload(LWCReloadEvent event) {
-        return;
-    }
-
-    @Override
-    public void load(LWC event) {
-        return;
     }
 
     @Override
@@ -94,46 +72,6 @@ public class LWCEvents implements Module {
     }
 
     @Override
-    public void onBlockInteract(LWCBlockInteractEvent event) {
-        return;
-    }
-
-    @Override
-    public void onCommand(LWCCommandEvent event) {
-        return;
-    }
-
-    @Override
-    public void onDestroyProtection(LWCProtectionDestroyEvent event) {
-        return;
-    }
-
-    @Override
-    public void onDropItem(LWCDropItemEvent event) {
-        return;
-    }
-
-    @Override
-    public void onEntityInteract(LWCEntityInteractEvent event) {
-        return;
-    }
-
-    @Override
-    public void onEntityInteractProtection(LWCProtectionInteractEntityEvent event) {
-        return;
-    }
-
-    @Override
-    public void onMagnetPull(LWCMagnetPullEvent event) {
-        return;
-    }
-
-    @Override
-    public void onPostRegistration(LWCProtectionRegistrationPostEvent event) {
-        return;
-    }
-
-    @Override
     public void onPostRemoval(LWCProtectionRemovePostEvent event) {
         Player player = event.getPlayer();
         if (player == null) {
@@ -157,16 +95,6 @@ public class LWCEvents implements Module {
     }
 
     @Override
-    public void onRedstone(LWCRedstoneEvent event) {
-        return;
-    }
-
-    @Override
-    public void onRegisterEntity(LWCProtectionRegisterEntityEvent event) {
-        return;
-    }
-
-    @Override
     public void onRegisterProtection(LWCProtectionRegisterEvent event) {
         Player player = event.getPlayer();
         if (player == null) {
@@ -174,11 +102,6 @@ public class LWCEvents implements Module {
         }
         Bookshelf.lwcCancelOpen.add(player.getUniqueId());
         Bukkit.getScheduler().runTaskLater(Bookshelf.plugin, () -> Bookshelf.lwcCancelOpen.remove(player.getUniqueId()), 5);
-    }
-
-    @Override
-    public void onSendLocale(LWCSendLocaleEvent event) {
-        return;
     }
 
 }

